room/views.py

```diff
@@ -1,17 +1,47 @@
+from typing import Any, Dict
+
 from rest_framework.views import APIView
 from rest_framework.response import Response
 
+from application.entities.room_entity import RoomEntity
+from application.repositories.interfaces import GetAllRooms
 from application.repositories.room_repository import RoomRepository
 from application.usecases.room_usecase import RoomUsecase
 
 from .repository import RoomDBRepository
 
 
-class RoomViewAPI(APIView):
-    def get(self, request):
-        db_room_repo = RoomDBRepository()
-        room_repo = RoomRepository(db_room_repo=db_room_repo)
-        room_usecase = RoomUsecase(room_repo=room_repo)
-        room_data = room_usecase.get_all_rooms()
+class RoomAPIView(APIView):
+    db_repo: RoomDBRepository = None
+    get_all_rooms: GetAllRooms = None
+    room_usecase: RoomUsecase = None
+
+    def __init__(
```

Comment:
This all adds a form of dependency injection to the view. Now we can test this view in isolation from the database (see view tests below).